lib/frosting/repository.rb

```diff
@@ -1,18 +1,45 @@
 require "active_support/core_ext/string/inflections"
+require "active_support/core_ext/module/delegation"
 
 module Frosting
   class Repository
+    def self.infer_presenter(resource)
+      "Presenters::#{resource.class.name}".constantize
+    end
+
+    def self.procify(arg)
+      arg.respond_to?(:call) ? arg : Proc.new { arg }
+    end
+
     def self.present(resource, options = {})
-      begin
-        klass = options[:presenter] || "Presenters::#{resource.class.name}".constantize
-        klass.new(resource, options[:context])
-      rescue LoadError
-        raise "No such presenter: #{klass}"
-      end
+      klass = options.fetch(:presenter) { infer_presenter(resource) }
+      klass = procify(klass).call(resource)
+      klass.new(resource, options.fetch(:context))
```

Comment:
Whereas the prior `options[:context]` effectively makes passing in a context optional (since it will resolve to `nil` if not present in the hash), I believe `options.fetch(:context)` [would raise a `KeyError`](http://ruby-doc.org/core-1.9.3/Hash.html#method-i-fetch), which would break existing client code.

In fact we might want to spec that `:context` is optional.